### PR TITLE
feat(policy): Add policy codes and tests for OpenEAI-VLA

### PR DESCRIPTION
**Pull request by ffllyy467**
## Summary / Motivation

This PR integrates **OpenEAI-VLA**, a Vision-Language-Action policy built on a Qwen3-VL backbone with a DiT action head trained via flow matching, into the LeRobot framework. The integration enables training, fine-tuning, and inference of OpenEAI-VLA within LeRobot's standard pipelines (`lerobot-train`, `lerobot-rollout`).

OpenEAI-VLA combines a frozen Qwen3-VL-4B-Instruct backbone with an 18-layer DiT (32 heads, 1664 hidden dim) that uses cross-attention to Qwen hidden states for action generation. Pretrained checkpoints are released at [`OpenEAI/OpenEAI-VLA-Pretrained`](https://huggingface.co/OpenEAI/OpenEAI-VLA-Pretrained) under the `openeai-lerobot/` subfolder. The integration supports per-dataset multi-subset branching (state/action encoders are auto-inferred from `data_dim_info` or checkpoint metadata), with automatic handling of dimension-mismatched encoders during fine-tuning. Multi-modal and multi-source data mixing from the original codebase are not yet ported and are noted as limitations in the docs.

## Paper / References

- **Paper**: [OpenEAI-Platform: An Open-source Embodied Artificial Intelligence Hardware-Software Unified Platform](https://arxiv.org/abs/2606.03392)
- **Original code**: [https://github.com/eai-yeslab/OpenEAI-VLA](https://github.com/eai-yeslab/OpenEAI-VLA)
- **Pretrained checkpoints**: [OpenEAI/OpenEAI-VLA-Pretrained](https://huggingface.co/OpenEAI/OpenEAI-VLA-Pretrained)

## Related issues

- Fixes / Closes: (none)
- Related: (none)

## What changed

- **New policy module**: added `src/lerobot/policies/openeai/` containing:
  - `configuration_openeai.py` — `OpenEAIVLAConfig` with default hyperparameters
  - `modeling_openeai.py` — `OpenEAIPolicy` and `OpenEAIModel` (Qwen3-VL + DiT)
  - `processor_openeai.py` — `Qwen3VLProcessor` wrapper for joint image+text processing
  - `blocks.py` — `DiTBlock`, `CrossAttention`, and helper layers
- **Policy registration**: registered `openeai` as a new policy type, selectable via `--policy.type=openeai` in `lerobot-train` / `lerobot-rollout`.
- **Documentation**: added a policy doc page describing architecture, configuration, pretrained checkpoint download, training / fine-tuning workflows, and limitations.
- **No breaking changes**: this is purely additive; no existing policies, configs, or APIs are modified.

### Tests added

New test files under `tests/policies/openeai/`:

- `test_openeai.py` — unit tests for `OpenEAIVLAConfig` validation, policy class registration, `CrossAttention` / `DiTBlock` / positional-embedding helpers, `DiTPolicyHead` forward and multi-subset branching, and state-dict round-trip determinism.
- `test_processor_openeai.py` — unit tests for the pre/post-processor pipeline construction, step ordering (`Rename -> Batch -> Qwen3VL -> Device -> Normalizer`), normalization mapping, empty-camera handling, and `Qwen3VLProcessorStep` helpers (`transform_features`, `_get_task`, `_detect_device`).
- `test_openeai_lerobot_vs_original.py` — parity tests against the original OpenEAI-VLA implementation: DiT head deterministic forward, cross-subset forward, flow-matching time sampling (Beta distribution), `noisy_action` / target-velocity construction, sin-cos / sinusoidal positional embeddings, timm attention mask correctness, and Euler-integration sample-action loop.

### How to run

Run the full OpenEAI policy test suite:

```bash
pytest -sv tests/policies/openeai/
```

### Quick reproduction for the reviewer

**1. Download the pretrained LeRobot-compatible checkpoint:**

```bash
hf download OpenEAI/OpenEAI-VLA-Pretrained \
  --include "openeai-lerobot/*" \
  --local-dir ./model/openeai-vla
```

**2. Train from scratch (sanity check, no pretrained weights):**

```bash
lerobot-train \
  --dataset.repo_id=${HF_USER}/your_dataset \
  --policy.type=openeai \
  --policy.qwen_path=Qwen/Qwen3-VL-4B-Instruct \
  --policy.freeze_backbone=true \
  --output_dir=outputs/train/openeai_scratch \
  --policy.device=cuda
```

**3. Fine-tune from the pretrained checkpoint (frozen backbone):**

```bash
lerobot-train \
  --dataset.repo_id=${HF_USER}/your_dataset \
  --policy.type=openeai \
  --policy.qwen_path=Qwen/Qwen3-VL-4B-Instruct \
  --policy.freeze_backbone=true \
  --policy.pretrained_path=./model/openeai-vla/openeai-lerobot \
  --output_dir=outputs/train/openeai_finetune \
  --policy.device=cuda
```

**4. Full fine-tuning (unfreeze backbone, requires more VRAM):**

```bash
lerobot-train \
  --dataset.repo_id=${HF_USER}/your_dataset \
  --policy.type=openeai \
  --policy.qwen_path=Qwen/Qwen3-VL-4B-Instruct \
  --policy.freeze_backbone=false \
  --policy.gradient_checkpointing=true \
  --policy.pretrained_path=./model/openeai-vla/openeai-lerobot \
  --output_dir=outputs/train/openeai_full_ft \
  --policy.device=cuda
```

**5. Rollout on a real robot (or any compatible eval setup):**

```bash
lerobot-rollout \
  --strategy.type=base \
  --policy.path=${HF_USER}/openeai_policy \
  --robot.type=so101_follower \
  --robot.port=/dev/ttyACM0
```

## Checklist (required before merge)

- [x] Linting/formatting run (`pre-commit run -a`)
- [x] All tests pass locally (`pytest`)
- [x] Documentation updated
- [ ] CI is green
- [x] Community Review: I have reviewed another contributor's open PR and linked it here: https://github.com/huggingface/lerobot/pull/3675

## Reviewer notes

- Anything the reviewer should focus on (performance, edge-cases, specific files) or general notes.
- Anyone in the community is free to review the PR.

### CODE REVIEW

**Comment by ffllyy467** [comment]
Hi @imstevenpmwork , apologize for ping - this is my first contribution and I wasn't sure who the right reviewer is, but the CI workflows are still awaiting approval. This PR adds OpenEAI-VLA policy support. Could you please approve them when you have a moment, or point me to the right person to review it? All local checks (pre-commit, pytest) are passing. Thanks!

**Comment by imstevenpmwork** [comment]
Hello @ffllyy467 

Thanks for contributing. Just a quick sanity test in here, did you make sure to follow the guidelines for contributing an in-tree policy to `lerobot` ? https://huggingface.co/docs/lerobot/v0.6.0/en/bring_your_own_policies#path-b-contributing-in-tree

Kind regards!

**Comment by ffllyy467** [comment]
Hello @imstevenpmwork, thanks for the review, and sorry for the delayed response!!

Yes, we followed the in-tree (Path B) guidelines. Concretely:

- **Policy module**: `src/lerobot/policies/openeai/` with `OpenEAIVLAConfig` (registered via `@PreTrainedConfig.register_subclass("openeai")`), `OpenEAIVLAPolicy(PreTrainedPolicy)`, and the `make_openeai_pre_post_processors` factory following the naming convention.
- **Wiring**: config re-exported in `policies/__init__.py`; `lerobot_modelcard_template.md` entry + `policy_docs` mapping; policy listed in the root README models table; `openeai` extra added in `pyproject.toml` and included in the `all` extra.
- **Docs**: `docs/source/openeai.mdx` (registered in `_toctree.yml`) plus the policy `README.md` symlink under `docs/source/policy_openeai_README.md`.
- **Dependency gating**: optional deps behind `TYPE_CHECKING`/`_transformers_available` + `require_package`; FSDP wrap units declared (`_fsdp_wrap_modules`).
- **Tests**: unit, processor, and parity tests under `tests/policies/openeai/`, all passing.
- **Checkpoint**: pretrained weights published at `OpenEAI/OpenEAI-VLA-Pretrained` (`openeai-lerobot/` subfolder).

One more point worth mentioning: OpenEAI-VLA is trained primarily on real-robot data - large-scale, real-world manipulation demonstrations - with the full pipeline (training data, pretraining process, and released checkpoints) public end-to-end (see the OpenEAI-Platform paper and the `OpenEAI/OpenEAI-VLA-Pretrained` repository). To our knowledge, this distinguishes it from the other VLA policies in the library, which typically open-source the code and fine-tuned checkpoints but do not release their pretraining data.

Thanks again!
